x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/profile/SuggestProfilesRequest.java

```diff
@@ -66,6 +82,95 @@ public ActionRequestValidationException validate() {
         if (size < 0) {
             validationException = addValidationError("[size] parameter cannot be negative but was [" + size + "]", validationException);
         }
+        if (hint != null) {
+            validationException = validateHint(validationException);
+        }
+        return validationException;
+    }
+
+    private ActionRequestValidationException validateHint(ActionRequestValidationException validationException) {
```

Comment:
Optional but feels like this belongs on the `Hint` class

Reply:
Agreed!